### ISSUE TITLE
False-positive "old-division" (W1619) on explicit float division

### ISSUE THREAD

**Issue by obi1kenobi** [PyCQA/pylint#3039]
My understanding is that W1619 (old-division) is supposed to check for [PEP238](https://www.python.org/dev/peps/pep-0238) safety, and help users avoid unfortunate surprises when division on integers becomes true division. However, I believe that if the number being divided is a float, PEP238 does not change the division semantics, and therefore the following code (call it `repro.py`) should run the same in both Python 2 and 3.

```python
x = float(4) / 2
print(x)
```

Possibly related to #2887 and #2803 

### Steps to reproduce
1. Create the `repro.py` file described above.
2. Run `pylint --reports=no --disable=all --enable=old-division repro.py`

### Current behavior
```bash
$ pylint --reports=no --disable=all --enable=old-division repro.py 
************* Module repro
repro.py:1:4: W1619: division w/o __future__ statement (old-division)

-----------------------------------
Your code has been rated at 5.00/10
```

### Expected behavior
Since the division is explicitly on a `float`-typed number, `pylint` reports no issues.

### pylint --version output
Both the newest 2.x and the newest 1.x seem to be affected:

```bash
$ pylint --version
pylint 2.3.1
astroid 2.2.5
Python 3.6.8 (default, Mar 14 2019, 21:00:19) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)]
```

```bash
$ pylint --version
No config file found, using default configuration
pylint 1.9.5, 
astroid 1.6.6
Python 3.6.8 (default, Mar 14 2019, 21:00:19) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)]
```

**Comment by PCManticore** [comment]
Thanks for the report, it definitely makes sense.

**Comment by bojanbg** [comment]
This still raises W1619 with pylint 2.4.4:
```python
print(100 * float(5) / 10)
```
My mental model was that pylint can recognize that the numerator is a float.
